microcosm_pubsub/handlers/uri_handler.py

```diff
@@ -42,121 +17,33 @@ class URIHandler(metaclass=ABCMeta):
 
-    There are five expected outcomes for this handler:
-      - Raising an error (e.g. a bug)
-      - Skipping the handlers (because the pubsub message carried enough information to bypass processing)
-      - Handling the message after fetching the resource by URI
-      - Ignore the message after fetching the resource by URI
-      - Raising a nack (e.g. because the resource was not committed yet)
-
-    The middle three cases are all handled here with the expectation that we produce *one* INFO-level
-    log per message processed (unless an error/nack is raised).
+    We still have the same five expected outcomes described in the base classe.
+    The only difference is that in this case skipping can happen either before fetching the resource
+    (using `get_reason_to_skip`) or afterwards.
 
     """
-    def __init__(
-        self,
-        graph,
-        retry_nack_timeout=1,
-        resource_nack_timeout=1,
-        resource_cache_enabled=True,
-        resource_cache_ttl=DEFAULT_RESOURCE_CACHE_TTL,
-        resource_cache_whitelist_callable=resource_cache_whitelist_callable,
-    ):
-        self.opaque = graph.opaque
-        self.retry_nack_timeout = retry_nack_timeout
-        self.resource_nack_timeout = resource_nack_timeout
-        self.resource_cache_ttl = resource_cache_ttl
-        self.resource_cache = self.get_resource_cache(graph) if resource_cache_enabled else None
-        self.resource_cache_whitelist_callable = resource_cache_whitelist_callable
-
-    @property
-    def nack_timeout(self):
-        """Deprecated, use retry_nack_timeout"""
-        return self.retry_nack_timeout
-
-    @property
-    def name(self):
-        return titleize(self.__class__.__name__)
-
-    @property
-    def nack_if_not_found(self):
-        return True
-
-    @property
-    def resource_type(self):
-        return dict
-
-    def get_resource_cache(self, graph):
-        try:
-            return graph.resource_cache
-        except (LockedGraphError, NotBoundError):
-            # Nb. if resource cache is globally disabled, will not be bound
-            return None
+    def __init__(self, graph, **kwargs):
+        super().__init__(graph, **kwargs)
+        self.fetch_resource = graph.config.sqs_message_dispatcher.fetch_uri_resource
 
     def __call__(self, message):
         uri = message["uri"]
-        self.on_call(message, uri)
-
-        skip_reason = self.get_reason_to_skip(message, uri)
-        if skip_reason is not None:
-            self.on_skip(message, uri, skip_reason)
-            return False
+        self._pre_handle(message, uri)
 
-        resource = self.convert_resource(
-            self.get_resource(message, uri),
-        )
+        resource = None
+        # XXX: remove conditional and use base class instead post-POC
+        if self.fetch_resource:
+            resource = self.convert_resource(
+                self.get_resource(message, uri),
+            )
```

Comment:
Those lines are the only difference with the base class for the `__call__` method.